assignments/array-methods.js

```diff
@@ -54,30 +56,99 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names 
+//into a new array called fullName. 
 let fullName = [];
-console.log(fullName);
+
+runners.forEach(function (runner) {
+
+    fullName.push(runner.first_name + " " + runner.last_name);
+  });
+  
+  console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. 
+//Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
 
+runners.map(function (runner, index) {
+
+    allCaps[index] = runner.first_name.toUpperCase();
+  });
+
+  console.log(allCaps);
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can 
+//choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log 
+//the result
 let largeShirts = [];
-console.log(largeShirts);
+
+largeShirts = runners.filter(function (runner) {
+
+    if (runner.shirt_size === 'L') {
+  
+      runners.filter(function (runner) {
+        return runner.shirt_size === 'L';
+      });
+      return runner;
+    }
+  });
+
+  console.log(largeShirts);
```

Comment:
.filter() returns an item from the original array based on a true or false statement: 
```
return runner.shirt_size === 'L';
```
is all you need in the function